package/network/services/hostapd/files/hostapd.sh

```diff
@@ -418,6 +419,25 @@ hostapd_set_psk() {
 	for_each_station hostapd_set_psk_file ${ifname}
 }
 
+hostapd_set_sae_password_file() {
+	local ifname="$1"
+	local vlan="$2"
+
+	json_get_vars mac vid key
+	local sae_password="${key}"
+	[ -n "$mac" ] && sae_password="${sae_password}|${mac}"
+	[ -n "$vid" ] && sae_password="${sae_password}|vlanid=${vid}"
+	echo "${sae_password}" >> /var/run/hostapd-${ifname}.sae_passwords
```

Comment:
It's a nitpick but because the key will be at the start of the line and it could be anything (including a `-`), I think a `printf '%s\n' "${sae_password}"` is more appropriate.